src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl

```diff
@@ -0,0 +1,178 @@
+"""
+    EllipticalSurface{T,TR,TP} <: AbstractSurfacePrimitive{T}
+
+* `r::TR`: 
+    * TR = Real -> Full Circle (a = b = r)
+    * TR = (Real, Real) -> Circular Annulus (r_in = r[1], r_out = r[2])
+    * TR = ((Real,), (Real,)) -> Full Ellipse (a = r[1][1], b = r[2][1])
+    * TR = ((Real, Real),(Real, Real)) -> Elliptical Annulus \n(a_in = r[1][1], a_out = r[1][2], b_in = r[2][1], b_out = r[2][2])
+    * Not all are implemented yet
+
+* `φ::TP`: 
+    * TP = Nothing <-> Full in φ
+    * ...
+"""
+@with_kw struct EllipticalSurface{T,TR,TP} <: AbstractPlanarSurfacePrimitive{T}
```

Comment:
Can we somehow constrain `TR` with respect to `T`, e.g. `where {T <: Union{T, Tuple{T,T},...}}`?

Reply:
Yes, we could. But it is optional and I was not sure in the beginning how it will look like exactly.